src/functions.js

```diff
@@ -609,14 +609,18 @@ var _ = Mavo.Functions = {
 	between: $.extend((haystack, from, to, tight) => {
 		[haystack, from, to] = [str(haystack), str(from), str(to)];
 
-		var i1 = from? haystack[tight? "lastIndexOf" : "indexOf"](from) : -1;
-		var i2 = haystack[tight? "indexOf" : "lastIndexOf"](to);
+		let fromIndex = haystack[tight? "lastIndexOf" : "indexOf"](from);
+		let toIndex = haystack[tight? "indexOf" : "lastIndexOf"](to);
 
-		if (from && i1 === -1 || i2 === -1) {
+		if (fromIndex === -1 || toIndex === -1) {
 			return "";
 		}
 
-		return haystack.slice(i1 + 1, i2 === -1 || !to? haystack.length : i2);
+		if (tight && toIndex <= fromIndex){
+			return haystack.slice(toIndex + to.length, fromIndex);
+		}
+
+		return haystack.slice(fromIndex + from.length, toIndex);
```

Comment:
But what if `from` and/or `to` are not provided? They are _optional_ by design (and you took it into account in your first commit). Moreover, the `fromLast()` and `toFirst()` functions are built on top of the optionality of `from` and `to`.

E.g., With this change, `fromlast("foo.bar.baz", ".")` returns `foo.bar` instead of `baz`, and `tofirst("foo.bar.baz", ".")` returns `bar.baz` instead of `foo`.

Reply:
fixed!

Reply:
Awesome! LGTM. Thanks!